### PR TITLE
BHoM: Issue 406 Added Duct and Node Objects

### LINKED ISSUE #406: Environment_oM: Add Duct and Node Objects
To add duct and node to Environments Object Model

### PR DESCRIPTION
**Pull request by kayleighhoude**
### Issues addressed by this PR
#406 

Closes #406

### CODE REVIEW

**Comment by michaldengusiak** [comment]
I will review from discipline content point of view and add @FraserGreenroyd to check compliance with environment structure

**Comment by ZiolkowskiJakub** [comment]
Hi,

I would like to chip in my comment:

Should we try to create more "abstract" object/interface to describe both Duct and Pipe? The equations for flow calculation will be quite similar. The difference is in fluid type.

**Comment by rwemay** [comment]
@michaldengusiak , agree. Great to see progress/a start here though. Perhaps we have a persistent branch for this (without pull requesting for a while) to get some parallel developing going quickly. A planning board with what functionality we're looking to try and achieve over what time frame would be good.

File: Environment_oM/Elements/Node.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Geometry;
+
+namespace BH.oM.Environment.Elements
+{
+    public class Node : BHoMObject, IElement0D
```
[the remark this answers is not in the document]
Comment (tg359) [reply Environment_oM/Elements/Node.cs]: I saw this issue and thought I should mention that the way EnergyPlus handles airflow between zones is to define a `Node` as a `Point` with an associated `Zone` to/from which air conditions will be obtained, and a type with properties:
- Inlet (generic supply point); or
- Return (generic return point); or
- Floor (supply/return point with additional propertieis associated with supplying air at low-level)
- Control (I have no idea what this is); or
- Ceiling (supply/return point with additional propertieis associated with supplying air at high-level); or
- MundtRoom (I have no idea what this is).

In the `EnergyPlus_Toolkit` there is currently a blank `AirNode` object ready for populating when we start doing natvent, but if something develops in this area I'll use whatever the outcome is.

Is there a method currently to find the containing zone for any/all objects in 3d-space? If not, we'd need one for this `Node` object.

[the remark this answers is not in the document]
Comment (FraserGreenroyd) [reply Environment_oM/Elements/Node.cs]: @tg359 could you post those comments on the issue (#406 ) as well? Just to further the discussion around there as well? :smile: Be good to combine and clarify the needs from an MEP perspective and EnergyPlus perspective/others as appropriate.

In reference, I do not believe we have a method to find the containing zone. That is something that would also need to be discussed further as part of an implementation plan for this.

>  - MundtRoom (I have no idea what this is).

I could make a few suggestions, but this is a public forum so I won't :wink:

**Comment by FraserGreenroyd** [comment]
Following discussion with @kayleighhoude , @al-fisher , @michaldengusiak yesterday this PR is closed.

**Comment by FraserGreenroyd** [comment]
@kayleighhoude I am not deleting the branch in case you want to keep working with what was here though I have saved a copy, but I'll leave you to delete the branch when you're happy you have everything :smile:

**Pull request closed without merging**